hpe3parclient/client.py

```diff
@@ -4439,3 +4439,141 @@ def targetInRemoteCopyGroupExists(
         except Exception:
             pass
         return False
+
+    def check_response(self, resp):
+        for r in resp:
+            if 'error' in str.lower(r) or 'invalid' in str.lower(r) or 'the schedule format' in str.lower(r):
+                err_resp = r.strip()
+                return err_resp
+
+    def createSchedule(self, schedule_name, task, taskfreq='@hourly'):
+        """Create Schedule for volume snapshot.
+        :param schedule_name - The name of the schedule
+        :type - string
+        :param task - command to for which schedule is created
+        :type - string
+        :param taskfreq - frequency of schedule
+        :type - string
+        """
+        cmd = ['createsched']
+        cmd.append("\""+task+"\"")
+        if '@' not in taskfreq:
+           cmd.append("\""+taskfreq+"\"")
+        else:
+           cmd.append(taskfreq)
+        cmd.append(schedule_name)
+        try:
+            resp = self._run(cmd)
+
+            err_resp = self.check_response(resp)
+            if err_resp:
+                raise exceptions.SSHException(err_resp)
+        except exceptions.SSHException as ex:
+            raise exceptions.SSHException(ex)
+
+
+    def deleteSchedule(self, schedule_name):
+        """Delete Schedule
+        :param schedule_name - The name of the schedule to delete
+        :type - string
+        """
+        cmd = ['removesched', '-f', schedule_name]
+        try:
+            resp = self._run(cmd)
+
+            err_resp = self.check_response(resp)
+            if err_resp:
+                err = (("Delete snapschedule failed Error is"
+                         " '%(err_resp)s' ") %
+                       {'err_resp': err_resp})
+                raise exceptions.HTTPNotFound(reason=err)
+        except exceptions.HTTPNotFound as ex:
```

Comment:
Since we are not making any HTTP REST call, this exception is not right.

We must have an SSH related exception wherever we are only making CLI call. Look at exception.py for an exception you can use, or else extend the SSHException to reflect the correct exception

This holds true everywhere we are generically throwing HTTPNotFound exception and no REST/HTTP calls are placed

Reply:
will do changes

Reply:
Just replacing a generic SSHException will not help. We need to be more specific when raising exception.

Look at exceptions.py and create the exception you need by extending SSHExceptions.

Reply:
should i use SSHException? or should i create DeleteScheduleException class in exception.py.
As per code , if Command fails it will be due to sshexception only